Sources/GRPC/GRPCClientStateMachine.swift

```diff
@@ -478,27 +493,32 @@ extension GRPCClientStateMachine.State {
 
   /// See `GRPCClientStateMachine.receiveEndOfResponseStream(_:)`.
   mutating func receiveEndOfResponseStream(
-    _ trailers: HTTPHeaders
+    _ trailers: HPACKHeaders
   ) -> Result<GRPCStatus, ReceiveEndOfResponseStreamError> {
-     let result: Result<GRPCStatus, ReceiveEndOfResponseStreamError>
-
-     switch self {
-     case .clientActiveServerActive,
-          .clientActiveServerIdle,
-          .clientClosedServerIdle,
-          .clientClosedServerActive:
-       result = .success(self.parseTrailers(trailers))
-       self = .clientClosedServerClosed
-
-     case .clientIdleServerIdle,
-          .clientClosedServerClosed:
+    let result: Result<GRPCStatus, ReceiveEndOfResponseStreamError>
+
+    switch self {
+    case .clientActiveServerIdle,
+         .clientClosedServerIdle:
+      result = self.parseTrailersOnly(trailers).map { status in
```

Comment:
What if we encounter an error here, I guess we don't need to transition our state in that case?

Reply:
I guess we could close at this point. As you mentioned, we just need to guarantee that everything gets shut down correctly and that the user gets notified (i.e. all necessary promises fulfilled).

Reply:
Exactly: the model here is fire an error. It's the role of the handler holding those promises to deal with that correctly (i.e. fail promises and close the channel).

Reply:
Sounds good! I guess this can be resolved then, sorry for the fuss.

Reply:
No problem, it's good to clarify these things!